### PR TITLE
AngularElement based on jQuery

### PR DESCRIPTION
**Pull request by svenwiegand**
`AngularElement` based on jQuery. When injecting `$element`, you can now set the type to `AngularElement` and get a lot of extra features.

### CODE REVIEW

**Comment by mysticfall** [comment]
I like the idea of introducing a static type for the native controller object. It made me wonder if we should  narrow the relevant types, like the return type of `ServiceProxy.proxy` as well?

I suppose `Directive.controller` could also benefit from this enhancement, because it's pretty unclear how to use it as it is.

Probably it might be reasonable to move all proxy(or 'wrapper', as you called it) related APIs into a single place, so I'd like to ask your opinion on this one.

Please let me know how you think about it. Thanks!

**Comment by mysticfall** [comment]
@svenwiegand I made some modifications to generalize the `ControllerWrapper` API, so it can be used for the other types as well (i.e. directives, filters, and etc).

Please let me know how you think about it.

**Comment by svenwiegand** [comment]
Sorry for the delay.

Not sure whether I've got the point of your changes (but didn't have time to fully check them out). The `ControllerWrapper` I've created was only necessary due to the fact, that in native AngularJS `Angular.element`'s `controller()` function directly returns the controller. In scalajs-angular I receive an object instead which has a `controller` property which then contains the controller. The wrapper was necessary to access the real controller through the `controller` property.

Regarding the `Directive.controller` property: In native AngularJS I've provided a constructor function here. This was great and easy, as it wasn't necessary to register the controller using a name with angular this way. Unfortunately in scalajs-angular I didn't find a way to provide a constructor function, so I've assigned a `String` to the `Directive.controller` property which contains the name of the (registered) controller to use. I could also use the `controller as`-syntax to avoid name clashes. So I've used either
```
class TabDirective ... {
  ...
  override def controller = Some(Angular.nameOf[TabController])
  ...
}
```
or
```
class TabDirective ... {
  ...
  override def controller = Some(Angular.nameOf[TabController] + " as tab")
  ...
}
```
though the string concatenation in the latter one really isn't nice.

Not sure whether this would work with your new implementation. Or what would be the correct way there to specify a controller for a directive?

**Comment by mysticfall** [comment]
I'm sorry that I could not check this project as often as I should, hopefully it goes back to normal soon so I won't have to stall RPs like that :sweat_smile: 

As you correctly pointed out, the object returned by this project cannot be used as an argument when a native Angular object is expected.

We had an unofficial API to deal with this problem before, as `Controller.proxy` method returned a native Angular object for the given _unregistered_ controller type or an instance.

It relied on the internal implementation detail, that every Scala objects created by this project holds a reference to the underlying native counterpart by its `controller` property.

Now as you have created a public trait to represent those native objects, I thought I could just generalize it a bit further after changing the property's name to `servive`, and narrow down the return type of the `directive.controller` method, which accepted those native objects before.

Actually I haven't realized you can return a controller name from that method before you mentioned, so I just changed it again to make it use a union type so both a constructor function or a name can be used.

To sum it up, now you can override the `Directive.controller` method and return a name as you did before, or use `Some(ServiceDefinition[YourController])` instead, in case you want to use an anonymous (unregistered) controller type as the directive's controller.

**Comment by svenwiegand** [comment]
No problem regarding the delay. I'm happy about receiving response at all...

Regarding the `Directive.controller` that really sounds great. Didn't like the usage of named controllers here at all. I'll try to switch the to anonymous controllers and see whether this works.

Thanks for your efforts...

**Comment by svenwiegand** [comment]
Just forgot to say, that to my opinion you should merge the changes from your branch.

**Comment by mysticfall** [comment]
Just merged the branch and published a new snapshot. Thanks again for the contribution :)
